apps/civicsignalblog/src/payload/access/canAccessApplication.js

```diff
@@ -1,7 +1,10 @@
-export default function canAccessApplication(user, searchString) {
+export default async function canAccessApplication(req, searchString) {
+  const { user, headers } = req;
+
   if (user) {
-    const app = user.currentApp || user.defaultApp;
-    return app === searchString.toLowerCase();
+    const currentApp = headers["x-current-app"] || user.currentApp;
+    const app = currentApp || user.defaultApp;
```

Comment:
```suggestion
    const app = headers["CS-App"] || user.currentApp || user.defaultApp;
```

One liner is still very legible, no?

The only thing I'm still not 100% happy here is APIs **MUST** provide `CS-App` header. You don't want a situation where we end up showing wrong data in the FE.